cli/root_test.go

```diff
@@ -42,52 +44,59 @@ func execute(cmd *cobra.Command, args ...string) (string, error) {
 }
 
 func stringContains(t *testing.T, testTarget, shouldBeHere string) {
+	t.Helper()
 	if !strings.Contains(testTarget, shouldBeHere) {
 		t.Errorf("Missing Content:\n   [%v]\nShould have been in here:\n   %v\n", shouldBeHere, testTarget)
 	}
 }
 
 func stringOmits(t *testing.T, testTarget, shouldNotBeHere string) {
+	t.Helper()
 	if strings.Contains(testTarget, shouldNotBeHere) {
 		t.Errorf("Extra Content that should not be found here:\n   [%v]\nBut it was:\n   %v", shouldNotBeHere, testTarget)
 	}
 }
 
 func stringChecks(t *testing.T, testTarget string, shouldBeHere, shouldNotBeHere []string) {
 	t.Helper()
-	expectedTermsMissing := []string{}
-	notExpectedTermsFound := []string{}
+
+	outputlogged := false
+	grouplogged := false
 
 	for _, v := range shouldBeHere {
 		if !strings.Contains(testTarget, v) {
-			expectedTermsMissing = append(expectedTermsMissing, v)
+			if !outputlogged {
+				outputlogged = true
+				t.Logf("String being checked:\n===========\n%v\n===========\n", testTarget)
+			}
+			if !grouplogged {
+				grouplogged = true
+				t.Logf("Content missing that was expected:\n")
+			}
+			t.Errorf("Should be (%v)\n", v)
 		}
 	}
+
+	grouplogged = false
+
 	for _, v := range shouldNotBeHere {
 		if strings.Contains(testTarget, v) {
```

Comment:
If replacing this with require.Contains is not possible, I would personally recommend altering the error messages to include both the comparison target and the changing value, which would eliminate these conditionals:

```go
// %q wraps the content in quotation marks
t.Errorf("string %q should not contain %q", testTarget, v)
```

Reply:
So the use case I am trying to accommodate is where the string under test is a long multiline string

Here is an example of code that would trigger an error:

```
func TestHelpCommand(t *testing.T) {

	stringChecks(t, execute(t, rootCmd, "help"),
		[]string{rootCmd.Long, "-s, --short-version", "-1, --oneline-version", "Francis", "Nickels"},
		[]string{"host"})
}
```

which generates:

```
--- FAIL: TestHelpCommand (0.01s)
    c:\git\src\github.com\fnickels\key-conjurer\cli\root_test.go:132: String being checked:
        ===========
        Key Conjurer retrieves temporary credentials from the Key Conjurer API.
        
        To get started run the following commands:
          keyconjurer login # You will get prompted for your AD credentials
          keyconjurer accounts
          keyconjurer get <accountName>
        
        Usage:
          keyconjurer [flags]
          keyconjurer [command]
        
        Available Commands:
          accounts           Prints the list of accounts you have access to.
          alias              Give an account a nickname.
          get                Retrieves temporary AWS API credentials.
          help               Help about any command
          identity-providers List identity providers you may use.
          login              Authenticate with KeyConjurer.
          roles              List all the roles that you can assume when using `keyconjurer get`.
          set                Sets config values.
          switch             Switch from the current AWS account into the one with the given Account ID.
          unalias            Remove alias from account.
          upgrade            Downloads the latest version of keyconjurer.
        
        Flags:
          -h, --help                         help for keyconjurer
              --host string                  The host of the KeyConjurer API
              --keyconjurer-rc-path string   path to .keyconjurerrc file (default "~/.keyconjurerrc")
          -1, --oneline-version              version for keyconjurer (single line format)
              --quiet                        tells the CLI to be quiet; stdout will not contain human-readable informational messages
          -s, --short-version                version for keyconjurer (short format)
        
        Use "keyconjurer [command] --help" for more information about a command.
        
        ===========
    c:\git\src\github.com\fnickels\key-conjurer\cli\root_test.go:132: Content missing that was expected:
    c:\git\src\github.com\fnickels\key-conjurer\cli\root_test.go:132: Should have found (Francis) in the output above
    c:\git\src\github.com\fnickels\key-conjurer\cli\root_test.go:132: Should have found (Nickels) in the output above
    c:\git\src\github.com\fnickels\key-conjurer\cli\root_test.go:132: Content found that was not expected:
    c:\git\src\github.com\fnickels\key-conjurer\cli\root_test.go:132: Should NOT have found (host) in the output above
FAIL
FAIL	github.com/riotgames/key-conjurer/cli	0.830s
FAIL
```

Reply:
Displaying the string under test multiple times would make it more difficult to consume the error report.  I am going for a listing that provides a display of the output buffer once and then iterates through all of the test conditions and only lists those that fail.